app/code/Magento/CatalogInventory/Model/Product/QuantityValidator.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * Copyright 2024 Adobe
+ * All Rights Reserved.
+ */
+declare(strict_types=1);
+
+namespace Magento\CatalogInventory\Model\Product;
+
+use Magento\CatalogInventory\Api\StockRegistryInterface;
+
+class QuantityValidator
+{
+    /**
+     * @var StockRegistryInterface
+     */
+    private $stockRegistry;
+
+    /**
+     * @param StockRegistryInterface $stockRegistry
+     */
+    public function __construct(
+        StockRegistryInterface $stockRegistry
```

Comment:
Use `readonly` instead. Please refer to [this](https://github.com/magento/magento2/blob/3428ebc3802a26fae9be542163579cefaa7967fa/app/code/Magento/Tax/Model/Plugin/AddTaxesExtensionAttribute.php#L46).

Reply:
done.